src/sagemaker/model.py

```diff
@@ -281,7 +284,11 @@ def __init__(
         self.model_data = model_data
         self.image_uri = image_uri
         self.predictor_cls = predictor_cls
-        self.env = env or {}
+        self.env = resolve_value_from_config(
+            env,
+            MODEL_PRIMARY_CONTAINER_ENVIRONMENT_PATH,
```

Comment:
how do we know this is the right key here? (aka today when env is provided, its used by the SDK in the same place in the API request object?)